afs/GettingStarted.md

```diff
@@ -0,0 +1,329 @@
+ - configure app file systems in the server powsybl configuration file, for instance defining a mapdb file system as above
+ - add `powsybl-afs-ws-client` to the runtime dependencies of your client application
+ - configure in the powsybl configuration of your application the following rest file system:
+```xml
+<rest-app-file-system>
+  <url-address>http://my-afs-server:8080/my-server-app</url-address>
+</rest-app-file-system>
+```
+
+Now all file systems defined in the server configurations will be transparently accessible from your client application, without changing any of your code!
+
+This allows for great flexibility in the deployment of your application, for instance to run the same application as standalone or client/server.
+
+**Note:**
+The underlying REST API is documented at http://my-afs-server:8080/my-server-app/rest/swagger
+
+### Listeners
+
+TO BE COMPLETED
+
+### Services
+
+TO BE COMPLETED
+
+### Dependencies
+
+TO BE COMPLETED
+
+# 2- Extending AFS for your needs
+
+### 2.1 Adding your own file types
+
+Powsybl-core already comes with a number of project file types:
+ - Imported cases and virtual cases
+ - Contingency stores
+ - Action scripts
+ - Security analysis runner: an object from which you can launch a security analysis
+
+However, you will probably want to extend that list with your own type of data or computation. AFS provides an easy way to do so.
+
+Each project file type in AFS relies on the definition of 3 classes:
+ - The actual new type, which must extends ProjectFile
+ - A builder class for the new type, which must extends ProjectFile
+ - An extension class, in charge of registering the new type and its builder to AFS. It must implement the ProjectFileExtension interface, and be registered through the use of `@AutoService` annotation.
+
+First define you new type :
+```java
+class FooFile extends ProjectFile {
+
+    FooFile(ProjectFileCreationContext context) {
+        super(context, 0);
+    }
+
+    public void doSomeCoolStuff() { ... }
+    public Foo getSomeCoolData() { ... }
+    public FooResult runSomeCoolComputation() { ... }
+}
+```
+
+Then define how to build it. This involves at least creating a new node in the underlying storage, and may involve writing some data blob or defining depencencies to other nodes:
+```java
+class FooFileBuilder implements ProjectFileBuilder<FooFile> {
+
+    private final ProjectFileBuildContext context;
+    private String name;
+
+    FooFileBuilder(ProjectFileBuildContext context) {
+        this.context = Objects.requireNonNull(context);
+    }
+
+    public FooFileBuilder withName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    //Storeds new node information in underlying storage, and builds actual object
+    @Override
+    public FooFile build() {
+        if (name == null) {
+            throw new IllegalStateException("name is not set");
+        }
```

Comment:
It could be interesting to define the concept of pseudoClass which is very important